Collections_HW/homework2/collections/DoublyLinkedList.java

```diff
@@ -0,0 +1,115 @@
+package homework2.collections;
+
+import java.util.Iterator;
+
+public class DoublyLinkedList implements Iterable<Student> {
+    private Node head;
+    private Node tail;
+    private int size;
+
+    private static class Node {
+        Student student;
+        Node next;
+        Node prev;
+
+        public Node(Student student, Node next, Node prev) {
+            this.student = student;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    // constructor to create a list
+    public DoublyLinkedList() {
+        head = null;
+        tail = null;
+        size = 0;
+    }
+
+    // inserts the student at the front of the list
+    public void push(Student student) {
+        Node newHead = new Node(student, head, null);
+        if (head != null) {
+            head.prev = newHead;
+        }
+        head = newHead;
+        if (tail == null) {
+            tail = newHead;
+        }
+        size++;
+    }
+
+    // removes and returns the first student of the list
+    public Student pop() {
+        if (size == 0) {
+            System.out.println("The list is empty. It's impossible to do this action now.");
+        }
+        Node oldHead = head;
+        head = head.next;
+        if (head != null)
+            head.prev = null;
+        size--;
+        return oldHead.student;
+    }
+
+    // appends the specified student to the end of the list
+    public boolean addLast(Student student) {
+        Node newTail = new Node(student, null, tail);
+        if (tail != null) {
+            tail.next = newTail;
+        }
+        tail = newTail;
+        if (head == null) {
+            head = newTail;
+        }
+        size++;
+        return true;
+    }
+
+    // removes and returns the last student
+    public Student removeLast() {
+        if (size == 0) {
```

Comment:
? return

Reply:
It always returns true after appending the element to the end of the list. Sorry, I don't get what is incorrect here.